### PR TITLE
Update documentation with upmerge changes

### PR DESCRIPTION
**Pull request by nordicjm**
Updates the release note documentation to have the zephyr commit
details resulting from the latest upmerge

### CODE REVIEW

**Review by nvlsianpu — COMMENTED** [review]
MCUboot changes:

boot_serial: refactoring and optimalization (mainly afects progresive erase implementation)
- migrate includes to <zephyr/...>
- switched DFU detection GPIO-pin configuration to be done through DTS.
  That pin is assigned thanks to `mcuboot-button0` pin-alias.
- boot_serial: Fix echo command code no longer compiling
- Added MCUboot status callback support, see `CONFIG_MCUBOOT_ACTION_HOOKS` Kconfig option.
- reworked LED usage, prefer DTS mcuboot-led0 alias over bootloader-led0 alias
- Added initial support for leverage RAM-LOAD mode with the zephyr-rtos port
- boot_serial: Upgrade from cddl-gen 0.1.0 to zcbor 0.4.0
- zephyr: removed device_get_binding() usage in favor to DEVICE_DT_GET()

- imgtool: Added support for providing the signature by 3rd party

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -41,6 +41,7 @@ Partition Manager
 * Added :kconfig:option:`CONFIG_PM_PARTITION_REGION_LITTLEFS_EXTERNAL`, :kconfig:option:`CONFIG_PM_PARTITION_REGION_SETTINGS_STORAGE_EXTERNAL`, and :kconfig:option:`CONFIG_PM_PARTITION_REGION_NVS_STORAGE_EXTERNAL` to specify that the relevant partition must be located in external flash memory.
   You must add a ``chosen`` entry for ``nordic,pm-ext-flash`` in your devicetree to make this option available.
   See :file:`tests/subsys/partition_manager/region` for example configurations.
+* Added :kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` to override the external driver check, this is needed when using an external flash which is not using the QSPI NOR driver
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:44]:
```suggestion
* Added :kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` to override the external driver check.
  This is needed when using an external flash which is not using the :ref:`QSPI NOR <zephyr:dtbinding_nordic_qspi_nor>` driver from Zephyr.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -726,40 +727,58 @@ Unity
...
+
+* Migrated includes to have ``zephyr/`` prefix
+
+* Switched DFU detection GPIO-pin configuration to be done through DTS using ``mcuboot-button0`` pin-alias.
+
+* Added MCUboot status callback support, see :kconfig:option:`CONFIG_MCUBOOT_ACTION_HOOKS`.
+
+* Reworked LED usage to prefer DTS ``mcuboot-led0`` alias over ``bootloader-led0`` alias
+
+* Added initial support for leveraging RAM-LOAD mode with the zephyr-rtos port
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:754]:
```suggestion
* boot_serial: 

  * Refactored and optimized the code, mainly in what affects the progressive erase implementation.
  * Fixed a compilation issue with the echo command code.
  * Upgraded from cddl-gen v0.1.0 to zcbor v0.4.0.

* imgtool: Added support for providing signature through a third party.
* zephyr:
   
  * Added initial support for leveraging the RAM-LOAD mode with the zephyr-rtos port.
  * Added the MCUboot status callback support.
    See :kconfig:option:`CONFIG_MCUBOOT_ACTION_HOOKS`.
  * Edited includes to have the ``zephyr/`` prefix.
  * Edited the DFU detection's GPIO-pin configuration to be done through DTS using the ``mcuboot-button0`` pin alias.
  * Edited the LED usage to prefer DTS' ``mcuboot-led0`` alias over the ``bootloader-led0`` alias.
  * Removed :c:func:`device_get_binding()` usage in favor of :c:func:`DEVICE_DT_GET()`.
```

File: scripts/partition_manager/partition_manager.rst [hunk 3]
```diff
@@ -523,6 +523,9 @@ After the ``nordic,pm-ext-flash`` value is set, you can place partitions in the
      region: external_flash
      size: CONFIG_EXTERNAL_PLZ_SIZE
 
+.. note::
+  If the external flash device is not using the QSPI NOR driver, :kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` must be enabled to override the PM external flash driver check, and the required driver must also be enabled for all applications that need it.
```
Comment (greg-fer) [inline scripts/partition_manager/partition_manager.rst:527]:
```suggestion
   If the external flash device is not using the :ref:`QSPI NOR <zephyr:dtbinding_nordic_qspi_nor>` driver, :kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` must be enabled to override the Partition Manager's external flash driver check, and the required driver must also be enabled for all applications that need it.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

**Review by greg-fer — APPROVED** [review]

**Comment by nordicjm** [comment]
@carlescufi @de-nordic can you review documentation updates please?

**Review by de-nordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**